T1429

secretrobotron opened the pull request.

jbuck committed:
Initial version of Dashboard. It doesn't actually work, and it looks pretty ugly [#1429]

secretrobotron committed:
[t1429] wip

1. unauthorized page
2. template filtering

secretrobotron committed:
[t1429] wip

Fiddling with template urls and savedDataURL

secretrobotron committed:
[t1429] savedDataUrls are correct(ish)

secretrobotron committed:
[t1429] tiny style adjustment

secretrobotron commented on cornfield/app.js:
```diff
@@ -117,101 +136,98 @@ function publishRoute( req, res ){
             projectData = JSON.parse( project.data ),
-            templateBase = VALID_TEMPLATES[ template ].replace( '{{templateBase}}', TEMPLATES_DIR + '/' );
-
-        fs.readFile( templateBase, 'utf8', function(err, conf){
```
NOTE: all of this is just an indentation change.

secretrobotron commented on cornfield/app.js:
```diff
@@ -230,7 +246,41 @@ function publishRoute( req, res ){
+          name: project.name,
+          template: project.template,
+          href: templateConfigs[ project.template ].template + 
+            "?savedDataUrl=" + PUBLISH_PREFIX + "/api/project/" + project._id
```
Here's the questionable meat.

jbuck replied:
It looks pretty good actually

secretrobotron committed:
[t1429] removed useless npm module, url

secretrobotron commented on cornfield/app.js:
```diff
@@ -14,6 +14,23 @@ const express = require('express'),
+  fs.readFile( configPath, 'utf8', function( err, conf ) {
+    var configPathBase = configPath.substring( 0, configPath.lastIndexOf( '/' ) );
+    conf = JSON.parse( conf );
+    conf.template = configPathBase + '/' + conf.template;
```
Attempted to use npm module `url` here, but was a bit wonky. File new ticket for follow up if desired, please.

jbuck commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/10a2d1cfa21276e/output.txt

jbuck commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/13d90380ae627ab/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/10a2d1cfa21276e/output.txt

Total script time: 0.48 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/13d90380ae627ab/output.txt

Total script time: 0.45 mins

jbuck commented:
/botio check

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @jbuck received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3c1cb84f47ba998/output.txt

jbuck commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @jbuck received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/05b59b0451aa221/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/3c1cb84f47ba998/output.txt

Total script time: 0.33 mins

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/05b59b0451aa221/output.txt

Total script time: 0.76 mins

Server is running on http://pr558.butyr.org/templates/test.html

secretrobotron committed:
[t1429] review changes

1. "Projects" button to send user to dashboard
2. id/name insertion on project loading from dashboard

humphd commented:
/botio preview

ghost commented:
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @humphd received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/b7b3f78e6e8cffd/output.txt

ghost commented:
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/b7b3f78e6e8cffd/output.txt

Total script time: 1.20 mins

Server is running on http://pr558.butyr.org/templates/test.html

jbuck committed:
Move default data to saved-data.json [#1429]
